script.js

```diff
@@ -1,4 +1,115 @@
+var currentGameMode = "waiting for player 1";
+var randomDiceRoll = [];
+
+var rollDiceOne = 0;
+var rollDiceTwo = 0;
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+  var userName = "";
```

Comment:
I presume you want to store the `userName` of both Player 1 and Player 2, and have their username values persist across the lifecycle of the application (ie. everytime I click Submit, it will not forget the usernames of Player 1 and Player 2)?

Consider using an array to store the usernames of Player 1 and Player 2. You have an [empty line on line 6](https://github.com/rocketacademy/basics-beat-that/blob/5f1d12f83849f7753b98e6098bd75b3b9e727abc/script.js#L6), so you can include this line in line 6:
```
var userNames = []
```
Then you can delete [line 9](https://github.com/rocketacademy/basics-beat-that/blob/5f1d12f83849f7753b98e6098bd75b3b9e727abc/script.js#L9), because it is now unnecessary - we will be using the array created earlier to store the values.

Since we are now using arrays, we can now replace [line 19](https://github.com/rocketacademy/basics-beat-that/blob/5f1d12f83849f7753b98e6098bd75b3b9e727abc/script.js#L19) with this:
```
if (userNames.length <= 0) {
  userNames.push(input);
}
```
The `if` condition basically checks if the array is empty (`.length` of less than or equal to 0). If it is empty, we assume we need to fill the array with Player 1's username. If it's not empty, we assume Player 1's username has been filled before, so we don't need to get Player 1's username from the input anymore.

On [line 24](https://github.com/rocketacademy/basics-beat-that/blob/5f1d12f83849f7753b98e6098bd75b3b9e727abc/script.js#L24), we want to print the username of Player 1. Remember that for arrays, the first item is in index 0 of the array. So, we will replace the entire myOutputValue re-assignment [from lines 22 to 30](https://github.com/rocketacademy/basics-beat-that/blob/5f1d12f83849f7753b98e6098bd75b3b9e727abc/script.js#L22-L30) with this:
```
// we use userNames[0] for Player 1's name
myOutputValue = myOutputValue =
      "Hello, " +
      userNames[0] +
      ". " +
      "You rolled " +
      Number(rollDiceOne) +
      " for Dice 1 and " +
      Number(rollDiceTwo) +
      " for Dice 2. <br>Please input 1 or 2 to indicate which number you would like to be the first number.";
```
We will need to change [line 56](https://github.com/rocketacademy/basics-beat-that/blob/5f1d12f83849f7753b98e6098bd75b3b9e727abc/script.js#L56) so that we are using the new `userNames` array created. Consider replacing line 56 this way:
```
// we only add the new username for player 2 if we have
// not previously added. we do so by checking the length
// of the array. if it's 1 or less, we assume player 2's
// name is not yet added
if (userNames.length <= 1) {
  userNames.push(input);
}
```
Your `else if` block for `"pick dice for player 1"` in [lines 33 to 50](https://github.com/rocketacademy/basics-beat-that/blob/5f1d12f83849f7753b98e6098bd75b3b9e727abc/script.js#L33-L50) can be changed, so that we do not tell the player(s) to `"Please enter player 2's name"` even when we are saving it from the start. You can replace the entirety of [lines 33 to 50](https://github.com/rocketacademy/basics-beat-that/blob/5f1d12f83849f7753b98e6098bd75b3b9e727abc/script.js#L33-L50) with the following (do read the comments!):
```
else if (currentGameMode == "pick dice for player 1") {
  console.log(currentGameMode);
  var feedbackText =  " Please enter player 2's name.";
  // if the length of the array is 2 or greater,
  // we assume we have already captured Player 1's and 2's names
  // thus, we can tell the player(s) to click Submit to continue the game
  if (userNames.length >= 2) {
    // userNames[1] refers to Player 2's name
    // remember that array indices are counted starting from 0
    enterPlayer2NameText = " Click Submit to start " + userNames[1] + "'s turn!"
  }
  
  if (input == 1) {
    currentGameMode = "waiting for player 2";
    myOutputValue =
      "Your final score is " +
      Number(String(rollDiceOne) + String(rollDiceTwo)) +
      feedbackText;
      randomDiceRoll.push(Number(String(rollDiceOne) + String(rollDiceTwo)));
  } else if (input == 2) {
    currentGameMode = "waiting for player 2";
    myOutputValue =
      "Your final score is " +
      Number(String(rollDiceTwo) + String(rollDiceOne)) +
      feedbackText;
    randomDiceRoll.push(Number(String(rollDiceTwo) + String(rollDiceOne)));
  }
}
``` 

Lastly, don't forget [line 57](https://github.com/rocketacademy/basics-beat-that/blob/5f1d12f83849f7753b98e6098bd75b3b9e727abc/script.js#L57). We are using the new array, and we want to say hi to Player 2. You can change line 57 to:
```
myOutputValue = "Hello, " + userNames[1];
```